sim/src/tlv.rs

```diff
@@ -564,6 +545,11 @@ impl ManifestGen for TlvGen {
             result.extend_from_slice(&buf);
         }
 
+        // Patch the size back into the TLV header.
+        let size = result.len() as u16;
```

Comment:
I believe this should discount the preprended protected TLVs, eg:

```
let size = (result.len() - protected_tlv.len()) as u16;
```
but since it passed the CI I might be wrong, or we might be missing some coverage.

Reply:
At least the size of the dependencies was included before these patches, although I think it was missing the size of the protected header. I think the correct answer, though, is to make the size `result.len() - size_pos - 2`, or something a little cleaner. So, it wasn't correct before, it is now incorrect in a different way, and apparently not checked.